### DISCUSSION TITLE
direkte InfluxDB Integration?

### DISCUSSION THREAD

**Discussion by dtechen** [tbnobody/OpenDTU#1554] [Ideas]
Hi,
das Projekt finde ich wirklich gut, was mir noch fehlte war die Integration nach (Cloud) InfluxDB. Daher habe ich das in einem Fork hinzugefügt. Hintergrund ist, dass ich eine Auswertung über einen längeren Zeitraum haben möchte, ohne eigene Infra zu betreiben (Cloud Influx -> Cloud Grafana). Gibt es Interesse daran? Das ist eine ernsthafte Frage, da man den Umweg über MQTT und Telegraf nehmen könnte.

**Comment by MichlW** [comment]
Ich habe es bei mir mit InfluxDB auf dem herkömmlichen Weg per MQTT über Telegraf umgesetzt und empfand es als Folter das aufzusetzen. Egal wie sehr ich mich bemüht habe die Logik der telegraf.conf zu verstehen, es gelang mir nicht. Am Ende kam ich zwar mit Trial and Error ans Ziel, aber das fühlt sich immernoch wackelig an. Und ich habe schon viele Projekte auf Linux mit diverser Software umgesetzt. Mit InfluxDB/FLUX selbst hab ich mich schneller angefreundet.
Grundsätzlich ließt sich das natürlich toll, direkt aus OpenDTU in eine InfluxDB schreiben zu können und **ich persönlich wäre dafür**.
Ich weiß aber nicht, ob das einer gängigen Methode widerspricht, wie mit InfluxDB umgegangen werden soll/darf.
Anderseits schreiben meine beiden AirRohr-Feinstaubsensoren auch direkt in die InfluxDB, wenn auch mit alter API.

**Comment by tm-107** [comment]
Interessiert wäre ich auch. Dann könnte der Umweg über ioBroker entfallen ...

**Comment by designermohr** [comment]
Oh. Wie schön von dem Thema InfluxDB + OpenDTU zu hören. Ohne dass ich von diesen o.g. Beiträgen etwas wusste: Ich bin seit ein Paar Tagen an einer Erstellung einer API-Anbindung der OpenDTU Daten aus dem JSON-File in eine InfluxDB (OSS 2 vs.2.7.1 als lokale Installation). 
Bisher arbeite ich mich durch 2 Import-Möglichkeiten (Telegraf Plugin: HTTP-JSON und HTTP) und habe bisher gute Fortschritte gemacht. Ok. InfluxDB ist in der Installation echt mühselig. Aber wenn man homebrew über den Terminal/Befehlseingabe nutzt und ein aktuelles OS hat, geht es recht zügig.
Die Leistungswerte eine InfluxDB sind phänomenal. Wahnsinnig schnell. Auch die Benutzer-Oberfläche ist anschaulich.
Ich bin noch nicht fertig. Sobald ich ein fertiges Skript habe, sodass alle relevanten Daten aus der OpenDTU via API in die InfluxDB problemlos geschrieben werden, melde ich mich hier noch einmal.

**Comment by maurosbicego** [comment]
Hallo zusammen,
erst einmal Danke für das super Projekt! Ich verwende OpenDTU seit einigen Tagen bei mir, und habe auch für andere Anwendungen in meinem Netzwerk eine InfluxDB und habe nun ein kleines Script gebastelt, welches die Daten regelmässig von OpenDTU abfragt und in ein Influx Bucket speichert (läuft in einem Docker). Das Ganze ist natürlich weniger performant als eine direkte Integration und evtl. noch buggy (habe es gerade vorhin geschrieben). Löst natürlich nicht das ursprüngliche Problem, da man eigene Infra braucht.
Falls jemand Interesse daran hat, lasse ich mal den Link hier liegen: https://github.com/maurosbicego/opendtu-to-influxdb
